src/lib/dovetail_audio.ts

```diff
@@ -0,0 +1,221 @@
+import {AdzerkRequest, AdzerkResponse} from './adzerk';
+import {AdzerkFetcher} from './adzerk_fetcher';
+import {ExtendableAudio} from '../javascript/extendable_audio';
+import {DovetailFetcher, DovetailResponse, ArrangementEntry} from './dovetail_fetcher';
+
+type AllUnion = [
+  DovetailResponse | PromiseLike<DovetailResponse>,
+  AdzerkResponse | PromiseLike<AdzerkResponse>
+];
+
+type AllResult = [DovetailResponse, AdzerkResponse];
+
+interface Arrangement {
+  entries: ArrangementEntry[];
+  duration?: number;
+}
```

Comment:
Our Arrangement gets replaced every time the overall URL is replaced, making this a good place to put cachable calculated values, like duration